pyrobosim/pyrobosim/utils/general.py

```diff
@@ -42,16 +42,25 @@ def __init__(self, filename=None):
         :param filename: Path to metadata YAML file.
         :type filename: str, optional
         """
+        self.data = {}
+        self.sources = []
```

Comment:
can we briefly explain what each of these attributes holds? Is `sources` the list of file paths?